test/e2e/user_workload_monitoring_test.go

```diff
@@ -893,50 +899,51 @@ func assertTenancyForRules(t *testing.T) {
 func assertTenancyForLabels(t *testing.T) {
 	const testAccount = "test-labels"
 
-	_, err := f.CreateServiceAccount(userWorkloadTestNs, testAccount)
+	err := framework.Poll(2*time.Second, 10*time.Second, func() error {
+		_, err := f.CreateServiceAccount(userWorkloadTestNs, testAccount)
+		return err
+	})
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	// Grant enough permissions to read labels.
-	_, err = f.CreateRoleBindingFromClusterRole(userWorkloadTestNs, testAccount, "admin")
+	err = framework.Poll(2*time.Second, 10*time.Second, func() error {
+		_, err = f.CreateRoleBindingFromClusterRole(userWorkloadTestNs, testAccount, "admin")
+		return err
+	})
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	var token string
-	err = framework.Poll(5*time.Second, 5*time.Minute, func() error {
+	err = framework.Poll(5*time.Second, time.Minute, func() error {
```

Comment:
just curious, how can we know the proper timeout value for an API call in Kubernetes?

Reply:
I'm reducing the number of retries because the test doesn't wait for a change to be propagated. Here the retries are needed to deal with transient API errors so a time out of 1 minute is more than enough.